src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -207,10 +209,13 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
                     );
                 });
 
+            // Fetch kernel the fastest possible way (local kernels from cache but remote fetch latest).
             // Fetch the list of kernels from the cache (note: if there's nothing in the case, it will fallback to searching).
+            // Fetching remote kernels cannot be done from cache.
             this.controllersPromise = Promise.all([
-                this.loadNotebookControllersImpl(true, 'useCache'),
-                this.loadNotebookControllersImpl(false, 'useCache')
+                this.loadNotebookControllersImpl(true, false, 'useCache'),
```

Comment:
Options instead of args would be nice to have here (so I know what 'true', 'false' is actually setting.)

Reply:
Agreed

Reply:
I'll do that seprately, found a few other things I'd like to clean up

Reply:
Never mind, done.